[Filebeat][Checkpoint] Fixing some field types and small typos

P1llus opened the pull request:
## What does this PR do?

This enhancement fixes a few small bugs or missing things:
Related users was missing.
Moving from JSON to YAML pipeline.
Duplicate entry for appending related.ip's.
Some source/destination fields was not mapped to client/server.
Fixed typo in one processor for observer interface
Cleaned up the client/server definition to rather just use source and destination, makes it much less lines of code and conditions.

While the changes seems a lot, it's pretty much just moving the code around to clean it up, fixing some small typos/bugs that came up and ensuring all field types are now set.

## Why is it important?

Fixes a few small quality of life settings and bugs before release

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Author's Checklist

Went through the pipeline and noted down all ECS related fields which is not string and added a converter processor for them.
Fixed a small typo and then reran nosetests to generate the new outcome, the new expected.json should show how the field types are now correct.

elasticmachine commented:
Pinging @elastic/siem (Team:SIEM)

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_18737.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #18737 opened]`

* Start Time: 2020-05-25T19:47:03.746+0000

* Duration: 45 min 26 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 2214  |
| Skipped      | 382 |
| Total        | 2596   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->

P1llus commented:
Since I moved a lot from set to rename it allowed me to remove a larger amount of if conditions (checking for null).

As with the fortinet comment, convert does not actually rename but copies, so for all fields converted I have added them to the remove processor array at the end.

elasticmachine commented:
## :broken_heart: Build Failed


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/2//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/2//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/2//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-18737/detail/PR-18737/2//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_18737.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #18737 updated]`

* Start Time: 2020-05-27T08:46:45.126+0000

* Duration: 41 min 48 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 1530  |
| Skipped      | 381 |
| Total        | 1911   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->


### Steps errors

  <details><summary>Expand to view the steps failures</summary>
  <p>
  
  * **Name**: `Mage update build test`
    * Description: mage  update build test
    
    * Duration: 7 min 26 sec
    
    * Start Time: 2020-05-27T09:09:21.838+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats-beats-mbp/pipelines/PR-18737/runs/2/steps/738/log)
    
  
  </p>
  </details>



<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-05-27T09:16:47.662Z] Recording test results
[2020-05-27T09:16:49.242Z] Stashed 2 file(s)
[2020-05-27T09:16:49.251Z] Archiving artifacts
[2020-05-27T09:16:50.096Z] + curl -sSLo codecov https://codecov.io/bash
[2020-05-27T09:16:50.370Z] + FILE=auditbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f auditbeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=filebeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f filebeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=heartbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f heartbeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=libbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f libbeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=metricbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f metricbeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=packetbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f packetbeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=winlogbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f winlogbeat/build/coverage/full.cov ]
[2020-05-27T09:16:50.370Z] + FILE=journalbeat/build/coverage/full.cov
[2020-05-27T09:16:50.370Z] + [ -f journalbeat/build/coverage/full.cov ]
[2020-05-27T09:16:51.638Z] Failed in branch Filebeat x-pack
[2020-05-27T09:17:41.923Z] >> build: Building filebeat
[2020-05-27T09:19:48.471Z] >> go test: Unit Testing
[2020-05-27T09:23:25.087Z] SUMMARY:
[2020-05-27T09:23:25.087Z]   Fail:     0
[2020-05-27T09:23:25.087Z]   Skip:     7
[2020-05-27T09:23:25.087Z]   Pass:     497
[2020-05-27T09:23:25.087Z]   Packages: 19
[2020-05-27T09:23:25.087Z]   Duration: 3m34.6219543s
[2020-05-27T09:23:25.087Z]   Coverage Report: C:\Users\jenkins\workspace\Beats_beats-beats-mbp_PR-18737\src\github.com\elastic\beats\x-pack\filebeat\build\TEST-go-unit.html
[2020-05-27T09:23:25.087Z]   JUnit Report:    C:\Users\jenkins\workspace\Beats_beats-beats-mbp_PR-18737\src\github.com\elastic\beats\x-pack\filebeat\build\TEST-go-unit.xml
[2020-05-27T09:23:25.087Z]   Output File:     C:\Users\jenkins\workspace\Beats_beats-beats-mbp_PR-18737\src\github.com\elastic\beats\x-pack\filebeat\build\TEST-go-unit.out
[2020-05-27T09:23:25.087Z] >> go test: Unit Test Passed
[2020-05-27T09:26:01.612Z] >> python test: Unit Testing
[2020-05-27T09:26:57.862Z] WARNING: You are using pip version 19.2.3, however version 20.1.1 is available.
[2020-05-27T09:26:57.862Z] You should consider upgrading via the 'python -m pip install --upgrade pip' command.
[2020-05-27T09:27:03.417Z] ........SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
[2020-05-27T09:27:03.417Z] [success] 15.61% test_http_endpoint.Test.test_http_endpoint_request: 0.8264s
[2020-05-27T09:27:03.417Z] [success] 13.17% test_http_endpoint.Test.test_http_endpoint_get_request: 0.6973s
[2020-05-27T09:27:03.417Z] [success] 12.38% test_http_endpoint.Test.test_http_endpoint_empty_body: 0.6555s
[2020-05-27T09:27:03.417Z] [success] 12.20% test_http_endpoint.Test.test_http_endpoint_wrong_content_header: 0.6462s
[2020-05-27T09:27:03.417Z] [success] 11.94% test_http_endpoint.Test.test_http_endpoint_malformed_json: 0.6324s
[2020-05-27T09:27:03.417Z] [success] 11.80% test_http_endpoint.Test.test_http_endpoint_missing_auth_value: 0.6247s
[2020-05-27T09:27:03.417Z] [success] 11.51% test_http_endpoint.Test.test_http_endpoint_wrong_accept_header: 0.6094s
[2020-05-27T09:27:03.417Z] [success] 11.39% test_http_endpoint.Test.test_http_endpoint_wrong_auth_value: 0.6030s
[2020-05-27T09:27:03.417Z] ----------------------------------------------------------------------
[2020-05-27T09:27:03.417Z] Ran 128 tests in 5.317s
[2020-05-27T09:27:03.417Z] 
[2020-05-27T09:27:03.417Z] OK (SKIP=120)
[2020-05-27T09:27:03.417Z] >> python test: Unit Testing Complete
[2020-05-27T09:27:03.493Z] Recording test results
[2020-05-27T09:27:08.500Z] Stashed 2 file(s)
[2020-05-27T09:27:08.510Z] Archiving artifacts
[2020-05-27T09:27:11.237Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats
[2020-05-27T09:27:11.550Z] + find . -type f -name TEST*.xml -path */build/* -delete
[2020-05-27T09:27:11.562Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats/Lint
[2020-05-27T09:27:11.638Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats/Filebeat-x-pack-Mac-OS-X
[2020-05-27T09:27:11.708Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats/Filebeat-x-pack
[2020-05-27T09:27:11.785Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats/Filebeat-x-pack-Windows
[2020-05-27T09:27:12.141Z] + cat
[2020-05-27T09:27:12.141Z] + /usr/local/bin/runbld ./runbld-script
[2020-05-27T09:27:12.141Z] Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
[2020-05-27T09:27:18.735Z] runbld>>> runbld started
[2020-05-27T09:27:18.735Z] runbld>>> 1.6.11/a66728ff8f4356963772e6e6d2069392fa06acbe
[2020-05-27T09:27:20.648Z] runbld>>> The following profiles matched the job 'Beats/beats-beats-mbp/PR-18737' in order of occurrence in the config (last value wins).
[2020-05-27T09:27:21.591Z] runbld>>> Debug logging enabled.
[2020-05-27T09:27:21.591Z] runbld>>> Storing result
[2020-05-27T09:27:21.852Z] runbld>>> Store result: created {:total 2, :successful 2, :failed 0} 1
[2020-05-27T09:27:21.852Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200527092721-58500A8E
[2020-05-27T09:27:21.852Z] runbld>>> Adding system facts.
[2020-05-27T09:27:22.799Z] runbld>>> Adding vcs info for the latest commit:  e0088c6f5c9234385fa7887c694232b8957b7195
[2020-05-27T09:27:23.059Z] runbld>>> >>>>>>>>>>>> SCRIPT EXECUTION BEGIN >>>>>>>>>>>>
[2020-05-27T09:27:23.060Z] runbld>>> Adding /usr/lib/jvm/java-8-openjdk-amd64/bin to the path.
[2020-05-27T09:27:23.060Z] Processing JUnit reports with runbld...
[2020-05-27T09:27:23.060Z] + echo 'Processing JUnit reports with runbld...'
[2020-05-27T09:27:23.321Z] runbld>>> <<<<<<<<<<<< SCRIPT EXECUTION END <<<<<<<<<<<<
[2020-05-27T09:27:23.321Z] runbld>>> DURATION: 13ms
[2020-05-27T09:27:23.321Z] runbld>>> STDOUT: 40 bytes
[2020-05-27T09:27:23.321Z] runbld>>> STDERR: 49 bytes
[2020-05-27T09:27:23.321Z] runbld>>> WRAPPED PROCESS: SUCCESS (0)
[2020-05-27T09:27:23.321Z] runbld>>> Searching for build metadata in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats
[2020-05-27T09:27:24.706Z] runbld>>> Storing build metadata: 
[2020-05-27T09:27:24.706Z] runbld>>> Adding test report.
[2020-05-27T09:27:24.707Z] runbld>>> Searching for junit test output files with the pattern: TEST-.*\.xml$ in: /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737/src/github.com/elastic/beats
[2020-05-27T09:27:26.101Z] runbld>>> Found 6 test output files
[2020-05-27T09:27:26.363Z] runbld>>> Test output logs contained: Errors: 0 Failures: 0 Tests: 1911 Skipped: 360
[2020-05-27T09:27:26.623Z] runbld>>> Storing result
[2020-05-27T09:27:26.623Z] runbld>>> FAILURES: 0
[2020-05-27T09:27:26.623Z] runbld>>> Store result: updated {:total 2, :successful 2, :failed 0} 2
[2020-05-27T09:27:26.623Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200527092721-58500A8E
[2020-05-27T09:27:26.623Z] runbld>>> Email notification disabled by environment variable.
[2020-05-27T09:27:26.623Z] runbld>>> Slack notification disabled by environment variable.
[2020-05-27T09:27:32.123Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-18737
[2020-05-27T09:27:32.225Z] [INFO] getVaultSecret: Getting secrets
[2020-05-27T09:27:32.264Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-05-27T09:27:32.940Z] + chmod 755 generate-build-data.sh
[2020-05-27T09:27:32.941Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-18737/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-18737/runs/2 FAILURE 2447552
[2020-05-27T09:27:32.941Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-18737/runs/2/steps/?limit=10000 -o steps-info.json
[2020-05-27T09:27:33.191Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-18737/runs/2/tests/?status=FAILED -o tests-errors.json
[2020-05-27T09:27:33.442Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-18737/runs/2/log/ -o pipeline-log.txt

```

</p>
</details>

andrewkroh commented:
jenkins, run tests please

andrewkroh commented:
The test failure looks like a transient issue building the pubsub emulator image. I'm re-running it.
